File: sdk/typescript/examples/07-volume-control.ts
// 07 — Volume control
//
// Read the current volume (with min/max range) and bump it up by one step.
//
//   bun run examples/07-volume-control.ts          # show + step up
//   bun run examples/07-volume-control.ts -3       # step down 3
//
// Volume is in firmware-defined steps (typically dB on PortalPlayer targets).

import { createClient } from './_client.js';

const client = createClient();
const delta = process.argv[2] ? Number(process.argv[2]) : 1;

const before = await client.sound.getVolume();
const range = before.max - before.min;
const filled = range > 0 ? Math.round(((before.volume - before.min) / range) * 20) : 0;
const bar = '█'.repeat(filled) + '░'.repeat(Math.max(0, 20 - filled));

console.log(`Volume: ${before.volume} dB  (range ${before.min} … ${before.max})`);
console.log(`        ${bar}`);

const after = await client.sound.adjustVolume(delta);
console.log(`\nAdjusted by ${delta >= 0 ? `+${delta}` : delta} → ${after} dB`);

File: sdk/elixir/examples/07_volume_control.exs
# 07 — Volume control
#
#   mix run examples/07_volume_control.exs

Code.require_file("_helper.exs", __DIR__)

client = Examples.Helper.client()

{:ok, vol} = Rockbox.Sound.volume(client)
IO.puts("Current: #{vol.volume} (range #{vol.min}..#{vol.max})")

{:ok, after_up} = Rockbox.Sound.up(client)
IO.puts("After +1: #{after_up}")

{:ok, _} = Rockbox.Sound.down(client)
IO.puts("Reverted.")
